android/kotlin/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/IngestionMetadataErichmentPlugin.kt

```diff
@@ -0,0 +1,25 @@
+package com.amplitude.ampli
+
+import com.amplitude.core.events.BaseEvent
+import com.amplitude.core.platform.Plugin
+
+class IngestionMetadataErichmentPlugin : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Enrichment
+    override lateinit var amplitude: com.amplitude.core.Amplitude
+
+    override fun execute(event: BaseEvent): BaseEvent? {
+        if (event.extra == null) event.extra = mapOf()
+        val ingestionMetadataObject =
+            IngestionMetadata("android-kotlin-ampli", "1.0.0").toJsonObject()
+        val extra = mapOf<String, Any>().plus(
+            mapOf(
+                "amplitude" to
+                        mapOf(
+                            "ingestionMetadata" to ingestionMetadataObject
```

Comment:
Do we need this class (`IngestionMetadata`) if we anyways need to covert it to json object? Think it might be too much to use the class -> object -> json object.

```
JSONObject("""{"source_name":"android-kotlin-ampli", "source_version":"1.0.0"}""")
```
or
```
val jsonObject = JSONObject()
rootObject.put("source_name","android-kotlin-ampli")
rootObject.put("source_version","1.0.0")
```

Reply:
I think we should use `IngestionMetadata` type. I wanted it to make setting the necessary explicit, and allow for better decoupling between Ampli and Amplitude SDK.

We should also probably add accessors for post-construction configuration. plugin.setIngestionMetadata(IngestionMetadata(...))